File: baseadapter/src/main/java/com/sum/simpleadapter/multiple/EntrustManager.java
package com.sum.simpleadapter.multiple;

import java.util.ArrayList;
import java.util.List;

import androidx.viewbinding.ViewBinding;

/**
 * @author liujiang
 * Desc:
 */
public class EntrustManager<T> {
    protected List<Entrust<ViewBinding, T>> container = new ArrayList<>();

    /**
     * 添加
     * @param item
     */
    public void add(Entrust<ViewBinding, T> item) {
        container.add(item);
    }

    /**
     * 获取当前的type
     * @param t
     * @return
     */
    public int getViewTypeByPosition(T t) {
        for (int i = 0; i < container.size(); i++) {
            Entrust<ViewBinding, T> item = container.get(i);
            if (item.isThisType(t)) {
                return i;
            }
        }
        return 0;
    }

    /**
     * 根据viewType获取child
     * @param viewType
     * @return
     */
    public Entrust<ViewBinding, T> getByViewType(int viewType) {
        return container.get(viewType);
    }

    /**
     * 获取当前对象的child
     * @param t
     * @return
     */
    public Entrust<ViewBinding, T> getByItem(T t) {
        return container.get(getViewTypeByPosition(t));
    }
} 

File: sampleAdapter/src/main/java/com/sum/sampleadapter/multiple/EntrustManager.kt
package com.sum.sampleadapter.multiple

import androidx.viewbinding.ViewBinding

/**
 * @author  LiuJiang
 * Desc:
 */
class EntrustManager<T> {
    private val container = mutableListOf<Entrust<*, T>>()

    /**
     * 添加
     */
    fun add(item: Entrust<*, T>) = container.add(item)

    /**
     * 获取当前的viewType类型
     */
    fun getViewType(t: T): Int {
        container.forEachIndexed { index, entrust ->
            if (entrust.isThisType(t)) return index
        }
        return 0
    }

    /**
     * 根据viewType获取ViewHolder
     */
    fun getViewHolderByViewType(viewType: Int): Entrust<ViewBinding, T> = container[viewType] as Entrust<ViewBinding, T>

    /**
     * 获取当前item获取ViewHolder
     */
    fun getViewHolderByItem(t: T): Entrust<ViewBinding, T> = container[getViewType(t)] as Entrust<ViewBinding, T>
}